nova/api.ts

```diff
@@ -0,0 +1,38 @@
+import { db } from './firebase';
+import express from 'express';
+import { Request, Response } from 'express';
+import { reduceMessage } from './MessageReducer'
+import { getFilePath } from './utils'
+
+const server = express();
+const router = express.Router();
+const PORT = 9000;
+
+const fs = require('fs');
+
+let getMembers = function () {
+    db.collection('members').get().then((vals) => {
+        vals.forEach((doc) => {
+            var data = doc.data();
+            fs.writeFile(getFilePath(data.email), JSON.stringify(data), function (err) {
+                if (err) {
+                    console.log(err);
+                }
+            });
+        });
+    })
+}
+
+router.post('/api/message', async (req: Request, res: Response) => {
+    let handled = await reduceMessage(req, res);
+    res.status(handled.status).json(handled);
+});
+
+server.use('/.netlify/functions/api', router);
+
+server.listen(PORT, () => {
+    console.log('IDOL scripts listening on port: ' + PORT);
+    getMembers();
+});
```

Comment:
why don't we use the same server in src?

Reply:
That's a good question! I think we wanted to keep all the functionality related to NOVA server different from that related to IDOL. I just edited console log statement to pretty convey that.

Reply:
I still don't get it why it's necessary. It seems to have a large overlap with the existing endpoints to support editing member info by web UI. To setup the new server, we need additional money for the server, additional build pipelines, and additional maintenance. I don't see how this additional service's benefit can justify the costs.